README.md
# Pialab-docker

## Purpose

This is for demo and dev only, please do not use it on production server !

## Requirements

Have docker installed and well configured on you favorite linux distribution

## Installation

```bash
git clone https://github.com/pia-lab/pialab-docker.git
cd pialab-docker
./install.sh

```

While it install everything, please be patient and have a drink

## Create a user in backend

Open firefox and go to http://localhost:8042/back

Login as lici@pialab.io / pia42

Create a new user (or not)

## Login in Pialab

Open firefox and go to http://localhost:8042/front

Login with the new created user (lici@pialab.io account will not work on the front part as it is an admin of the back)

Start using Pialab (or not)

## Please report any bug to :

https://github.com/pia-lab/pialab and/or https://github.com/pia-lab/pialab-back

## Environement Variable

Some variable change the installation process

### ${NAME}

Default NAME=pialab

It used for various isolation tricks, like container name, database user and more ...

### ${BRANCH}

Default BRANCH=master

It the git branch where to get pialab and pialab-back.
It can also be a tag like : BRANCH=0.4.2

### ${BUILDENV}

Default BUILDENV="dev"

It is the environement for build symfony and angular app.

It can be set to "prod" to change installation command behaviour like "ng build" or "bin/console cache:clear" ...

### ${ETCDDATA}

Default ETCDDATA="$(pwd)/var/${NAME}-etcd-data"

Local directory where etcd data are stored . (You should not need it)

### ${POSTGRESDATA}

Default  POSTGRESDATA="$(pwd)/var/${NAME}-postgres-data"

Local directory where postgresql data are stored. (You may need it if you want to save or move them)

### ${SERVERPORT}

Default SERVERPORT=8042

The docker rediction to container port 80

### ${FRONTURL}

Default FRONTURL="http://localhost:${SERVERPORT}/front"

The pialab url, it used when pialba-back build as paramater.

It must be the future url where you will use the front office app

### ${BACKURL}

Default BACKURL="http://localhost:${SERVERPORT}/back"

The pialab-back url, it used when pialba build as paramater.

It must be the future url where you will use the back office app

### ${CREATEUSER}

Default CREATEUSER=true

Used to enable or disable default user creation (lici@pialab.io and api@pialab.io).

It is usefull to set it to false if you want to re-use the same postgresql database several time (to avoid try to create several time the same user)

Warning if you set it to false you will have to connect to the docker container and create user with symfony command

### Example

```bash
export SERVERPORT=8042
export FRONTURL=http://demo.pialab.io:$SERVERPORT/front
export BACKURL=http://demo.pialab.io:$SERVERPORT/back
export NAME=demo.pialab
export ETCDDATA=$(pwd)/var/etcd-data
export POSTGRESDATA=$(pwd)/var/postgres-data
export CREATEUSER=true
export BUILDENV=prod
git clone https://github.com/pia-lab/pialab-docker.git
cd pialab-docker
./install.sh

```


CHANGELOG.md
# Change Log

## [1.1.0](https://github.com/pia-lab/pialab-docker/tree/1.1.0) (2018-06-20)
[Full Changelog](https://github.com/pia-lab/pialab-docker/compare/1.0.1...1.1.0)

## [1.0.1](https://github.com/pia-lab/pialab-docker/tree/1.0.1) (2018-06-14)
[Full Changelog](https://github.com/pia-lab/pialab-docker/compare/0.4.5...1.0.1)

**Implemented enhancements:**

- request for authentication : security problem [\#5](https://github.com/pia-lab/pialab-docker/issues/5)

**Fixed bugs:**

-  input fields  problem [\#3](https://github.com/pia-lab/pialab-docker/issues/3)

**Closed issues:**

- issues for dpo validation [\#4](https://github.com/pia-lab/pialab-docker/issues/4)
- login problem with user on front  [\#2](https://github.com/pia-lab/pialab-docker/issues/2)
- seizure problem [\#1](https://github.com/pia-lab/pialab-docker/issues/1)

## [0.4.5](https://github.com/pia-lab/pialab-docker/tree/0.4.5) (2018-05-18)
[Full Changelog](https://github.com/pia-lab/pialab-docker/compare/0.4.4...0.4.5)

## [0.4.4](https://github.com/pia-lab/pialab-docker/tree/0.4.4) (2018-05-16)
[Full Changelog](https://github.com/pia-lab/pialab-docker/compare/0.4.3...0.4.4)

## [0.4.3](https://github.com/pia-lab/pialab-docker/tree/0.4.3) (2018-05-14)
[Full Changelog](https://github.com/pia-lab/pialab-docker/compare/0.4.2...0.4.3)

## [0.4.2](https://github.com/pia-lab/pialab-docker/tree/0.4.2) (2018-05-11)


\* *This Change Log was automatically generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator)*

install.sh
#!/usr/bin/env bash

set -ex



if [ -z "${NAME}" ]
then
    NAME=pialab
fi

if [ -z "${BRANCH}" ]
then
   BRANCH=1.1.0
fi

if [ -z "${BUILDENV}" ]
then
    BUILDENV="dev"
fi

mkdir -p $(pwd)/var

if [ -n "${ETCDDATA}" ]
then
    ETCDARG="--volume ${ETCDDATA}:/etcd-data"
fi

if [ -n "${POSTGRESDATA}" ]
then
    POSTGRESARG="--volume ${POSTGRESDATA}:/var/lib/postgresql/data"
fi

if [ -z "${SERVERPORT}" ]
then
    SERVERPORT=8042
fi

if [ -z "${FRONTURL}" ]
then
    FRONTURL="http://localhost:${SERVERPORT}/front"
fi

if [ -z "${BACKURL}" ]
then
    BACKURL="http://localhost:${SERVERPORT}/back"
fi

if [ -z "${CREATEUSER}" ]
then
    CREATEUSER=true
fi

# clean
docker rm -f etcd.${NAME}.cnt || echo "Ok"
docker rm -f postgresql.${NAME}.cnt || echo "Ok"
docker rm -f apache.${NAME}.cnt || echo "Ok"
docker network rm ${NAME}.network || echo "Ok"

# create network
docker network create ${NAME}.network

# install etcd
REGISTRY=quay.io/coreos/etcd
docker run -dt --network=${NAME}.network \
       $ETCDARG \
       --name etcd.${NAME}.cnt ${REGISTRY}:v3.1.1 \
       /usr/local/bin/etcd \
       --data-dir=/etcd-data --name node1 \
       --initial-advertise-peer-urls http://0.0.0.0:2380 \
       --listen-peer-urls http://0.0.0.0:2380 \
       --advertise-client-urls http://0.0.0.0:2379 \
       --listen-client-urls http://0.0.0.0:2379

ETCDHOST=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' etcd.${NAME}.cnt)

# install postgres
DBROOTUSER=postgres
DBROOTPASSWORD=postgres24
docker run -dt --network=${NAME}.network \
       $POSTGRESARG \
       -e POSTGRES_PASSWORD=${DBROOTPASSWORD} \
       --name postgresql.${NAME}.cnt postgres:9.6

DBHOST=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' postgresql.${NAME}.cnt)

echo 'var' > .dockerignore

# install ${NAME}
docker rm -f apache.${NAME}.cnt|| echo "Ok"
docker -D build --network=${NAME}.network --build-arg CACHEBUST=$(shuf -n 1 -i 100-1000) \
       --build-arg DBHOST=${DBHOST} --build-arg ETCDHOST=${ETCDHOST} \
       --build-arg FRONTURL=${FRONTURL} --build-arg BACKURL=${BACKURL} \
       --build-arg CREATEUSER=${CREATEUSER} \
       --build-arg BRANCH=${BRANCH} --build-arg NAME=${NAME} \
       --build-arg BUILDENV=${BUILDENV} \
       -f Dockerfile -t ${NAME}.img .
docker run -dt --network=${NAME}.network -p ${SERVERPORT}:80 --name apache.${NAME}.cnt ${NAME}.img
